kedro-telemetry/tests/integration/test_telemetry.py

```diff
@@ -0,0 +1,82 @@
+from pytest import fixture
+import pytest
+
+
+@fixture
+def dummy_project_path():
+    return Path(__file__).parent / "dummy-project"
+
+
+class TestKedroTelemetryHookIntegration:
+    def test_telemetry_sent_once_with_kedro_run(self, mocker, dummy_project_path):
+        mocked_heap_call = mocker.patch("kedro_telemetry.plugin._send_heap_event")
+        mocker.patch(
+            "kedro_telemetry.plugin._check_for_telemetry_consent", return_value=True
+        )
+        kedro_cli = KedroCLI(dummy_project_path)
+        CliRunner().invoke(kedro_cli, ["run"])
+        mocked_heap_call.assert_called_once()
+
+    def test_telemetry_sent_once_with_other_kedro_command(
+        self, mocker, dummy_project_path
+    ):
+        from kedro_telemetry.plugin import telemetry_hook
+
+        telemetry_hook.consent = None
+        telemetry_hook._sent = False
+        telemetry_hook.event_properties = None
+        telemetry_hook.project_path = None
+
+        mocked_heap_call = mocker.patch("kedro_telemetry.plugin._send_heap_event")
+        mocker.patch(
+            "kedro_telemetry.plugin._check_for_telemetry_consent", return_value=True
+        )
+        kedro_cli = KedroCLI(dummy_project_path)
+        CliRunner().invoke(kedro_cli, ["run"])
+        mocked_heap_call.assert_called_once()
+
+    def test_telemetry_sent_once_with_session_run(self, mocker, dummy_project_path):
+        from kedro_telemetry.plugin import telemetry_hook
+
+        telemetry_hook.consent = None
+        telemetry_hook._sent = False
+        telemetry_hook.event_properties = None
+        telemetry_hook.project_path = None
+
+        mocked_heap_call = mocker.patch("kedro_telemetry.plugin._send_heap_event")
+        mocker.patch(
+            "kedro_telemetry.plugin._check_for_telemetry_consent", return_value=True
+        )
+        # Mock because all tests are sharing the kedro_telemetry.plugin.telemetry_hook object
+
+        bootstrap_project(dummy_project_path)
+        with KedroSession.create(project_path=dummy_project_path) as session:
+            session.run()
+        mocked_heap_call.assert_called_once()
+
+    def test_telemetry_sent_multi_with_other_kedro_command(
+        self, mocker, dummy_project_path
+    ):
+        from kedro_telemetry.plugin import telemetry_hook
+
+        telemetry_hook.consent = None
+        telemetry_hook._sent = False
+        telemetry_hook.event_properties = None
+        telemetry_hook.project_path = None
+
+        mocked_heap_call = mocker.patch("kedro_telemetry.plugin._send_heap_event")
+        mocker.patch(
+            "kedro_telemetry.plugin._check_for_telemetry_consent", return_value=True
+        )
+        kedro_cli = KedroCLI(dummy_project_path)
+        CliRunner().invoke(kedro_cli, ["run"])
+        CliRunner().invoke(kedro_cli, ["run"])
+        CliRunner().invoke(kedro_cli, ["run"])
+        mocked_heap_call.assert_called_once()
+        raise ValueError("It should fail!")
```

Comment:
```suggestion
    def test_telemetry_sent_multi_with_other_kedro_command(
        self, mocker, dummy_project_path
    ):
        from kedro_telemetry.plugin import telemetry_hook

        telemetry_hook.consent = None
        telemetry_hook._sent = False
        telemetry_hook.event_properties = None
        telemetry_hook.project_path = None

        mocked_heap_call = mocker.patch("kedro_telemetry.plugin._send_heap_event")
        mocker.patch(
            "kedro_telemetry.plugin._check_for_telemetry_consent", return_value=True
        )
        kedro_cli = KedroCLI(dummy_project_path)
        CliRunner().invoke(kedro_cli, ["run"])
        CliRunner().invoke(kedro_cli, ["run"])
        CliRunner().invoke(kedro_cli, ["run"])
        mocked_heap_call.assert_called_once()
        raise ValueError("It should fail!")
```

I take the liberty to add this test directly into the PR since I think it's easier to show. This test should fail but passing for some reason

Reply:
I see, the intend of having duplicate `CliRunner().invoke(kedro_cli, ["run"])` is to stimulate `kedro run` multiple times (expected multiple event).

One immediate thought came to mind, does that mean if someone is doing multiple `session.run` in a notebook, we only receive the 1st telemetry event?

Reply:
Thanks, @noklam . I tried running a session multiple times in a notebook, but I didn’t know that it’s actually prohibited (see screen). When I reloaded Kedro using the `%reload_kedro` command, the next `session.run()` did send data to telemetry, so it looks like we don’t have this issue in real usage. It seems to only be a problem in the testing environment. Therefore, I propose we remove the last test: `test_telemetry_sent_multi_with_other_kedro_command`.
<img width="1160" alt="Screenshot 2024-09-23 at 11 06 01" src="https://github.com/user-attachments/assets/472c52ee-ecd5-4fed-a938-7fc1b35dd575">